lib/travis/logs/drain_consumer.rb

```diff
@@ -95,7 +95,7 @@ def dead?
       ensure
         @dead = true
         @batch_buffer = nil
-        sleep
+        exit
```

Comment:
i'd be curious to see if this exits the whole process.

```
require 'thread'

Thread.new {
  exit
}

puts "sup"
sleep 3
puts "done"
```

this experiment suggests it would, but there could be something different about how this is set up in drain.